django/transformer/transform/fhir.py

```diff
@@ -63,19 +57,16 @@ def build_fhir_object(row, path_attributes_map, index=None):
             # If we didn't find an index in the path, then we don't worry about arrays
             val = row[attr.path]
 
-            if isinstance(val, (tuple, list)) and len(val) == 1:
-                # If we have a value with length 1, we simply extract the value and put
-                # it in the fhir object
-                insert_in_fhir_object(fhir_object, path, val[0])
-            elif isinstance(val, (tuple, list)) and index is not None:
+            if isinstance(val, list) and index is not None:
                 # If index is not None, we met an array before. Here, val will have
                 # several elements but we know which one we need
-                insert_in_fhir_object(fhir_object, path, val[index])
+                data_value = val[0] if len(val) == 1 else val[index]
```

Comment:
isn't index equal to 0 if len(val) is 1 ?

Reply:
Not necessarily, in some cases, we could want to fill an array of objects were some fields come a join (with several elements) and some fields that will only have 1 element in the array